core/src/com/unciv/models/ruleset/unique/IHasUniques.kt

```diff
@@ -42,4 +42,15 @@ interface IHasUniques : INamed {
 
     fun hasUnique(uniqueType: UniqueType, stateForConditionals: StateForConditionals? = null) =
         getMatchingUniques(uniqueType.placeholderText, stateForConditionals).any()
+
+    fun techsRequiredByUniques(): Sequence<String> {
+        val uniquesForWhenThisIsAvailable: Sequence<Unique> = getMatchingUniques(UniqueType.OnlyAvailableWhen, StateForConditionals.IgnoreConditionals)
```

Comment:
I think this would look more readable as a chain, eg a.b{}.c{}
What do you think?

Reply:
Better?

I want to keep the listing of the Uniques separate, since there's some fiddly technical detail happening there (fiddly to me, anyway) in handling the StateForConditionals, but the rest looks simple as a chain.